### PR TITLE
feat: allow access to SSL attributes in EL TemplateEngine

### PR DESCRIPTION
**Pull request by phiz71**
gravitee-io/issues#5322

### CODE REVIEW

File: src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java [hunk 1]
```diff
@@ -45,18 +40,45 @@ public static void main(String[] args) throws IOException {
     }
 
     public void run() throws IOException {
-        Map<String, Object> all = new HashMap<>();
+        Map<String, Object> all = new TreeMap<>();
 
-        Map<String, HashMap<String, Object>> request = new HashMap<>();
+        Map<String, TreeMap<String, Object>> request = new TreeMap<>();
         for (Method declaredMethod : EvaluableRequest.class.getDeclaredMethods()) {
```
Comment (marcambier) [inline src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java:46]: suggestion : should we mutualize some duplicated code to improve readability ?

For example : `Collection<Method> getGettersOfClass(Class<> classz)`
Would avoid 3 identation levels

**Review by gcusnieux — APPROVED** [review]

File: src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java (hunk 1 above)
Comment (phiz71) [reply src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java:46]: The code is almost the same but it has some differences that prevent it from being shared.

File: src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java [hunk 2]
```diff
@@ -45,18 +40,45 @@ public static void main(String[] args) throws IOException {
     }
 
     public void run() throws IOException {
-        Map<String, Object> all = new HashMap<>();
+        Map<String, Object> all = new TreeMap<>();
```
Comment (kamiiiel) [inline src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java:43]: Is there any reason you chose TreeMap instead of HashMap?

Comment (phiz71) [reply src/main/java/io/gravitee/gateway/api/el/EvaluableExtractor.java:43]: Yes, TreeMap allows me to easily sort Values by keys.
And the EvaluableExtractor is just a helper class used to generate a json file.
It is not needed at runtime.

**Pull request merged**